### PR TITLE
#2485. Add constant evaluation tests for `Set`

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: Language/Expressions/Constants/constant_set_t01.dart [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/// The potentially constant expressions and constant expressions are the
+/// following:
+/// ...
+/// • A constant set literal, const <T>{e1, ..., en}, or <T>{e1, ..., en} that
+///   occurs in a constant context, is a potentially constant expression if T is
+///   a constant type expression, and e1, . . . , en are constant expressions.
+///   It is further a constant expression if the set literal evaluates to an
+///   object.
+///
+/// @description Checks that a constant set literal of the for
```
Comment (eernstg) [inline Language/Expressions/Constants/constant_set_t01.dart:19]: Typo?
```suggestion
/// @description Checks that a constant set literal of the form
```

File: Language/Expressions/Constants/constant_set_t02.dart [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/// following:
+/// ...
+/// • A constant set literal, const <T>{e1, ..., en}, or <T>{e1, ..., en} that
+///   occurs in a constant context, is a potentially constant expression if T is
+///   a constant type expression, and e1, . . . , en are constant expressions.
+///   It is further a constant expression if the set literal evaluates to an
+///   object.
+///
+/// @description Checks that it is a compile-time error if in a constant set
+/// literal of the form const `<T>{e1, ..., en}`, or `<T>{e1, ..., en}` `T` is
```
Comment (eernstg) [inline Language/Expressions/Constants/constant_set_t02.dart:20]: Typo:
```suggestion
/// literal of the form const `<T>{e1, ..., en}`, or `<T>{e1, ..., en}`, `T` is
```

File: Language/Expressions/Constants/constant_set_t03.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+/// following:
+/// ...
+/// • A constant set literal, const <T>{e1, ..., en}, or <T>{e1, ..., en} that
+///   occurs in a constant context, is a potentially constant expression if T is
+///   a constant type expression, and e1, . . . , en are constant expressions.
+///   It is further a constant expression if the set literal evaluates to an
+///   object.
+///
+/// @description Checks that it is a compile-time error if in a constant set
+/// literal of the form const `<T>{e1, ..., en}`, or `<T>{e1, ..., en}` any of
```
Comment (eernstg) [inline Language/Expressions/Constants/constant_set_t03.dart:20]:
```suggestion
/// literal of the form const `<T>{e1, ..., en}`, or `<T>{e1, ..., en}`, any of
```

**Review by eernstg — COMMENTED** [review]
Looks good, just a couple of typos.

**Comment by sgrekhov** [comment]
Thank you! Typos are fixed. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**